src/server/main.py

```diff
@@ -157,39 +156,110 @@ def _get_siblings(params, headers):
 def _search_taxa(params, headers):
```

Comment:
Not really being used anywhere correctly IIRC but I tried to fix it anyway. Uses same new query as `search_species`